File: xperiment/hurufBergantian.c
#include <stdio.h>
int main(){
  char str[]="";
  gets(str);
  char char_stored;
  int count;
  for(int i=0;i<=strlen(str);i++){
    char chr=str[i];
    if(chr==char_stored){
      count+=1;
    }else if(chr=="P"){
      char_stored=chr;
    }else if(chr=="C"){
      char_stored=chr;
    }
  }
  printf("count: %i\n",count);
}

File: xperiment/hurufBergantian.js
//var str = 'PCPCPCPCPCPCPPC'
const readline = require('readline-sync');
var str = readline.question("input: ");

function hurufBergantian(string) {
  let count = 0;
  let char_stored = '';
  for(let i=0;i<=string.length;i++) {
    char = string.charAt(i);
    if(char == char_stored) count+=1;
    else if(char == 'P') char_stored = char;
    else if(char == 'C') char_stored = char;
  }
  return count;
}

console.log('output:', hurufBergantian(str));

